src/main/java/de/tum/in/www1/artemis/service/ComplaintResponseService.java

```diff
@@ -42,54 +42,68 @@ public ComplaintResponseService(ComplaintRepository complaintRepository, Complai
             throw new BadRequestAlertException("The complaint you are referring to does already have a response", ENTITY_NAME, "complaintresponseexists");
         }
-
         Result originalResult = originalComplaint.getResult();
         User assessor = originalResult.getAssessor();
         User reviewer = this.userService.getUser();
-
         StudentParticipation studentParticipation = (StudentParticipation) originalResult.getParticipation();
         if (!isUserAuthorizedToRespondToComplaint(studentParticipation, originalComplaint, assessor, reviewer)) {
             throw new AccessForbiddenException("Insufficient permission for creating a complaint response");
         }
-
-        originalComplaint.setAccepted(complaintResponse.getComplaint().isAccepted());
-        originalComplaint = complaintRepository.save(originalComplaint);
-
-        complaintResponse.setSubmittedTime(ZonedDateTime.now());
+        complaintResponse.setCreatedTime(ZonedDateTime.now());
         complaintResponse.setReviewer(reviewer);
-        // make sure the original complaint from the database is connected to the complaint response as we take it out later one and
-        // potential changes on the client side (e.g. remove student id) should not be saved
         complaintResponse.setComplaint(originalComplaint);
         return complaintResponseRepository.save(complaintResponse);
     }
 
+    public ComplaintResponse updateComplaintResponse(ComplaintResponse updatedComplaintResponse) {
+        if (updatedComplaintResponse.getId() == null) {
+            throw new BadRequestAlertException("To update a complaint response it needs an id", ENTITY_NAME, "idmissing");
+        }
+        Optional<ComplaintResponse> originalComplaintResponseOptional = complaintResponseRepository.findById(updatedComplaintResponse.getId());
+        if (originalComplaintResponseOptional.isEmpty()) {
+            throw new BadRequestAlertException("The complaint response you are referring to does not exist", ENTITY_NAME, "complaintresponsemissing");
+        }
+        ComplaintResponse originalComplaintResponse = originalComplaintResponseOptional.get();
+        Optional<Complaint> originalComplaintOptional = complaintRepository.findByIdWithEagerAssessor(originalComplaintResponse.getComplaint().getId());
+        if (originalComplaintOptional.isEmpty()) {
+            throw new BadRequestAlertException("The complaint you are referring to does not exist", ENTITY_NAME, "complaintmissing");
+        }
+        Complaint originalComplaint = originalComplaintOptional.get();
+
+        Result originalResult = originalComplaint.getResult();
+        User assessor = originalResult.getAssessor();
+        User reviewer = this.userService.getUser();
+        StudentParticipation studentParticipation = (StudentParticipation) originalResult.getParticipation();
+        if (!isUserAuthorizedToRespondToComplaint(studentParticipation, originalComplaint, assessor, reviewer)) {
```

Comment:
Does this enforce the lock?

Reply:
The lock feature is not implemented yet.